src/amber/cli/config.cr

```diff
@@ -1,28 +1,57 @@
+require "yaml"
+
 module Amber::CLI
   def self.config
     if File.exists? AMBER_YML
       Config.from_yaml File.read(AMBER_YML)
     else
       Config.new
     end
+  rescue ex : YAML::ParseException
+    logger.error "Couldn't parse #{AMBER_YML} file", "Watcher", :red
+    exit 1
+  end
+
+  def generate_config
+    File.write(AMBER_YML, CLI.config.to_yaml)
   end
 
   class Config
-    property database : String = "pg"
-    property language : String = "slang"
-    property model : String = "granite"
-    property recipe : (String | Nil) = nil
-    property recipe_source : (String | Nil) = nil
+    alias Watch = Hash(String, Hash(String, Array(String)))
+
+    getter database : String = "pg"
+    getter language : String = "slang"
+    getter model : String = "granite"
+    getter recipe : String?
+    getter recipe_source : String?
+    getter watch : Watch = {
+      "server" => {
+        "files" => [
+          "src/**/*.cr",
+          "src/**/*.#{@language}",
+          "config/**/*.cr"
+        ],
+        "commands" => [
+          "shards build -p --no-color",
+          "bin/#{app_name}"
+        ]
+      }
+    }
 
     def initialize
```

Comment:
If I remove the initialize then I can't use `Config.new` because it complains about no matching parameters :sweat_smile: 

Let me add a comment about this :+1: